Correctly detect noqa when used in method calls

### Problem

At rotki we have places where the `noqa: E501` rule comes in after or in function declarations like https://github.com/rotki/rotki/blob/535b7995dbf65f2824fac6fb8b73c1e1f2350613/rotkehlchen/db/upgrades/v33_v34.py#L113

### Example

In those places flake correctly detects the rule to skip the error but ruff doesn't. For a reproducible scenario I have 

```python
def caller(string: str) -> None:
    assert string is not None


caller(
    """
    This is test for a very long string where flake would prevent raising the error sicne we use noqa but ruff doesn't
    """,  # noqa: E501
)
```

That has this two different outputs

```
(rotki-main) ➜  rotki git:(trueblocks) ✗ flake8 ruff_test.py
(rotki-main) ➜  rotki git:(trueblocks) ✗ ruff ruff_test.py 
Found 1 error(s).
ruff_test.py:7:1: E501 Line too long (118 > 100 characters)
(rotki-main) ➜  rotki git:(trueblocks) ✗ 
```

### Expected behaviour

ruff should detect that the noqa rule is being used to skip the long line

--- a/resources/test/fixtures/E501.py
+++ b/resources/test/fixtures/E501.py
@@ -14,3 +14,21 @@ Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod tempor i
 
 _ = "---------------------------------------------------------------------------AAAAAAA"
 _ = "---------------------------------------------------------------------------亜亜亜亜亜亜亜"
+
+
+def caller(string: str) -> None:
+    assert string is not None
+
+
+caller(
+    """
+    Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod tempor incididunt ut labore et dolore magna aliqua.
+    """,
+)
+
+
+caller(
+    """
+    Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod tempor incididunt ut labore et dolore magna aliqua.
+    """,  # noqa: E501
+)

--- a/src/noqa.rs
+++ b/src/noqa.rs
@@ -1,4 +1,3 @@
-use std::cmp::{max, min};
 use std::collections::{BTreeMap, BTreeSet};
 use std::fs;
 use std::path::Path;
@@ -46,38 +45,17 @@ pub fn extract_noqa_directive(line: &str) -> Directive {
 
 pub fn extract_noqa_line_for(lxr: &[LexResult]) -> Vec<usize> {
     let mut noqa_line_for: Vec<usize> = vec![];
-
-    let mut min_line = usize::MAX;
-    let mut max_line = usize::MIN;
-    let mut in_string = false;
-
     for (start, tok, end) in lxr.iter().flatten() {
         if matches!(tok, Tok::EndOfFile) {
             break;
         }
-
-        if matches!(tok, Tok::Newline) {
-            min_line = min(min_line, start.row());
-            max_line = max(max_line, start.row());
-
-            // For now, we only care about preserving noqa directives across multi-line
-            // strings.
-            if in_string {
-                for i in (noqa_line_for.len())..(min_line - 1) {
-                    noqa_line_for.push(i + 1);
-                }
-                noqa_line_for.extend(vec![max_line; (max_line + 1) - min_line]);
+        if matches!(tok, Tok::String { .. }) {
+            for i in (noqa_line_for.len())..(start.row() - 1) {
+                noqa_line_for.push(i + 1);
             }
-
-            min_line = usize::MAX;
-            max_line = usize::MIN;
-        } else {
-            min_line = min(min_line, start.row());
-            max_line = max(max_line, end.row());
+            noqa_line_for.extend(vec![end.row(); (end.row() + 1) - start.row()]);
         }
-        in_string = matches!(tok, Tok::String { .. });
     }
-
     noqa_line_for
 }
 

